packages/execution-environments/src/common/validation.ts

```diff
@@ -41,3 +65,138 @@ export function validateExport(type: HandlerType, snapExport: unknown) {
   const validationFunction = VALIDATION_FUNCTIONS[type];
   return validationFunction?.(snapExport) ?? false;
 }
+
+export const JsonRpcRequestWithoutIdStruct = assign(
+  omit(JsonRpcRequestStruct, ['id']),
+  object({
+    id: optional(JsonRpcIdStruct),
+  }),
+);
+
+export type JsonRpcRequestWithoutId = Infer<
+  typeof JsonRpcRequestWithoutIdStruct
+>;
+
+export const EndowmentStruct = string();
+export type Endowment = Infer<typeof EndowmentStruct>;
+
+/**
+ * Check if the given value is an endowment.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is an endowment.
+ */
+export function isEndowment(value: unknown): value is Endowment {
+  return is(value, EndowmentStruct);
+}
+
+/**
+ * Check if the given value is an array of endowments.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is an array of endowments.
+ */
+export function isEndowmentsArray(value: unknown): value is Endowment[] {
+  return Array.isArray(value) && value.every(isEndowment);
+}
+
+const OkStruct = literal('OK');
+
+export const PingRequestArgumentsStruct = optional(
+  union([literal(undefined), array()]),
```

Comment:
Why do we need a union if it is optional?

Reply:
I think I ran into some issues otherwise, and needed this to keep it compatible.